### PR TITLE
Auto deploy SNAPSHOT versions.

### LINKED ISSUE #512: Auto-deployed SNAPSHOTs
Maybe I'm missing something but it would be really beneficial to have auto-deployed snapshots to verify certain newly added behaviors without the need to create another RC.

### PR DESCRIPTION
**Pull request by vanniktech**
Fixes #512. Hopefully. I've got some questions.

### CODE REVIEW

File: .buildscript/deploy_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  echo "Skipping snapshot deployment: wrong repository. Expected '$SLUG' but was '$TRAVIS_REPO_SLUG'."
+elif [ "$TRAVIS_JDK_VERSION" != "$JDK" ]; then
+  echo "Skipping snapshot deployment: wrong JDK. Expected '$JDK' but was '$TRAVIS_JDK_VERSION'."
+elif [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  echo "Skipping snapshot deployment: was pull request."
+elif [ "$TRAVIS_BRANCH" != "$BRANCH" ]; then
+  echo "Skipping snapshot deployment: wrong branch. Expected '$BRANCH' but was '$TRAVIS_BRANCH'."
+else
+  echo "Deploying snapshot..."
+  ./gradlew bintrayUpload
```
Comment (vanniktech) [inline .buildscript/deploy_snapshot.sh]: Is that the right task? You'll need to set the environment variable in Travis for the Username + password.

You can encrypt them though - https://docs.travis-ci.com/user/encrypting-files/#using-gpg

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-detektVersion=1.0.0.RC8
+detektVersion=1.0.0.RC9-SNAPSHOT
```
Comment (vanniktech) [inline gradle.properties]: I hope that's the right one I changed and RC9-SNAPSHOT sounds good

Comment (arturbosch) [reply gradle.properties]: yes its good :)

File: .buildscript/deploy_snapshot.sh (hunk 1 above)
Comment (arturbosch) [reply .buildscript/deploy_snapshot.sh]: Yes, its the right one. Hm but bintray will tell me that I already have that version when uploading the same version twice

Comment (arturbosch) [reply .buildscript/deploy_snapshot.sh]: I think you need to add `override = true` to the bintray closure.

Comment (vanniktech) [reply .buildscript/deploy_snapshot.sh]: We should only do that for snapshots though, right?

Comment (arturbosch) [reply .buildscript/deploy_snapshot.sh]: Yeah, lets check if the vesion endswith SNAPSHOT then?

Comment (vanniktech) [reply .buildscript/deploy_snapshot.sh]: Yup. Done that now.

**Comment by vanniktech** [comment]
@arturbosch ready for another review.

**Pull request merged**